app/src/main/res/layout/fragment_home.xml

```diff
@@ -3,140 +3,148 @@
+                style="@style/text_view_home_title"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_below="@+id/recyclerviewCourseContinue"
+                android:ellipsize="end"
+                android:maxLines="2"
+                android:text="@string/upcoming_class" />
+
+            <TextView
+                android:id="@+id/viewAll"
+                style="@style/text_view_home_title"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_below="@+id/recyclerviewCourseContinue"
+                android:layout_toStartOf="@+id/ivArrow"
+                android:text="@string/view_all"
+                android:visibility="visible" />
+
+            <androidx.appcompat.widget.AppCompatImageView
+                android:id="@+id/ivArrow"
+                android:layout_width="@dimen/dimen_30_dp"
+                android:layout_height="@dimen/dimen_30_dp"
+                android:layout_below="@+id/recyclerviewCourseContinue"
+                android:layout_alignParentEnd="true"
+                android:layout_marginTop="@dimen/dimen_10_dp"
+                android:padding="@dimen/dimen_5_dp"
+                android:src="@drawable/ic_right" />
+
+            <LinearLayout
+                android:id="@+id/ll"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_below="@+id/upcomingClass"
+                android:orientation="vertical">
+
+                <ProgressBar
+                    android:id="@+id/progress_bar_button_my"
+                    style="?android:attr/progressBarStyleSmall"
+                    android:layout_width="@dimen/dimen_progress_bar"
+                    android:layout_height="@dimen/dimen_progress_bar"
+                    android:layout_gravity="center"
+                    android:indeterminate="true"
+                    android:visibility="gone"
+                    app:elevation="2dp" />
+
+                <androidx.recyclerview.widget.RecyclerView
+                    android:id="@+id/recyclerview_my_upcoming_class"
+                    android:layout_width="wrap_content"
+                    android:layout_height="@dimen/height_rv_upcoming_class"
+                    android:layout_margin="@dimen/dimen_5_dp"
+                    android:orientation="horizontal"
+                    app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+                    tools:listitem="@layout/item_upcoming_class" />
+
+                <include
+                    android:id="@+id/emptyMyClass"
+                    layout="@layout/component_discover_class"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_marginBottom="@dimen/dimen_5_dp"
```

Comment:
my suggestion would be to start using dimen values that follow multiplier nomenclature (i.e. spacing_1x etc) instead of hardcoded name, would help us modify these spacings in future. This could be replaced with spacing_1x